### DISCUSSION TITLE
Help with Rocket v0.5.0-rc.2 and Docker

### DISCUSSION THREAD

**Discussion by physics515** [rwf2/Rocket#2257] [Questions] [answered]
Excuse me if I'm doing something stupid (this is my first Docker project, and I'm fairly new to rust) I'm having an issue getting Rocket 5 rc.2 working with Docker. When running the container, I don't get any output whatsoever and it doesn't seem to be working at all. See below for my setup:

**main.rs**
```
#[macro_use]
extern crate rocket;

#[launch]
async fn rocket() -> _ {
	rocket::build()
        .mount("/", routes![index])
}

#[get("/")]
pub(crate) async fn index() -> String {
    "Hello, world!".to_string()
}
```

**Cargo.toml**
```
[package]
name = "rocket-docker-test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
rocket = { git = "https://github.com/SergioBenitez/Rocket", rev = "4dcd928" }
```

**Rocket.toml**
```
[default]
address = "0.0.0.0"
port = 80

[release]
log_level = "debug"
```

Dockerfile
```
FROM rust:1.61 as builder
RUN USER=root

RUN mkdir rocket-docker-test
WORKDIR /rocket-docker-test
ADD . ./
RUN cargo clean && cargo build --release

FROM debian:bullseye
ARG APP=/user/src/app
RUN mkdir -p {$APP}

# Copy the compiled binaries into the new container.
COPY --from=builder /rocket-docker-test/target/release/rocket-docker-test ${APP}/rocket-docker-test

WORKDIR ${APP}

EXPOSE 80

CMD ["./rocket-docker-test"]
```

**Docker commands**

`docker build -t rocket-docker-test . `
`docker run -dp 8000:80  rocket-docker-test`

**Docker Logs**
![image](https://user-images.githubusercontent.com/7359379/176914237-b61ee03b-bd95-432a-b138-d1e186d3769a.png)

**Test**
![image](https://user-images.githubusercontent.com/7359379/176914832-26a126c2-d9ef-40db-aadb-7f63173f6905.png)

**Comment by physics515** [comment, marked as answer]
Solution from reddit user that worked.

The Rocket.toml file must be copied over to the final container with the binary.

**Working Dockerfile**
```
FROM rust:1.61 as builder
RUN USER=root

RUN mkdir rocket-docker-test
WORKDIR /rocket-docker-test
ADD . ./
RUN cargo clean && cargo build --release

FROM debian:bullseye
ARG APP=/user/src/app
RUN mkdir -p {$APP}

# Copy the compiled binaries into the new container.
COPY --from=builder /rocket-docker-test/target/release/rocket-docker-test ${APP}/rocket-docker-test
COPY --from=builder /rocket-docker-test/Rocket.toml ${APP}/Rocket.toml

WORKDIR ${APP}

EXPOSE 80

CMD ["./rocket-docker-test"]
```

**Comment by etsea117** [comment]
@physics515 did you encounter any issues with mounting the static files?

**Reply by physics515** [reply]
I'm not sure what you are referring to exactly. I assume you mean static files that would be served. In that case, no, at least once I figured out a little more about docker.

The only other major problem I really ran into was with TLS, I'm hosting on Azure so I'm having to convert the .p12 cert to .pem on spin-up.

**Reply by etsea117** [reply]
Ah, I've been struggling with getting docker to load the static folder in the same way that the non-containerized version of the rocket app finds it.

**Reply by physics515** [reply]
Are you copying the static file over? If I could see you Dockerfile I could maybe help.
